nodebuilder/share/module.go

```diff
@@ -34,7 +34,75 @@ func ConstructModule(tp node.Type, cfg *Config, options ...fx.Option) fx.Option
 		fx.Supply(*cfg),
 		fx.Error(cfgErr),
 		fx.Options(options...),
-		fx.Provide(newModule),
+		fx.Provide(newShareModule),
+		shrexSubComponents(),
+	)
+
+	bridgeAndFullComponents := fx.Options(
+		fx.Provide(getters.NewStoreGetter),
+		discoveryComponents(cfg),
+		shrexServerComponents(cfg),
+		edsStoreComponents(cfg),
+		fullAvailabilityComponents(),
+		shrexGetterComponents(cfg),
+		fx.Provide(func(shrexSub *shrexsub.PubSub) shrexsub.BroadcastFn {
+			return shrexSub.Broadcast
+		}),
+	)
+
+	switch tp {
+	case node.Bridge:
+		return fx.Module(
+			"share",
+			baseComponents,
+			bridgeAndFullComponents,
+			fx.Provide(func() peers.Parameters {
+				return cfg.PeerManagerParams
+			}),
+			fx.Provide(peers.NewManager),
+			fx.Provide(bridgeGetter),
+			fx.Invoke(func(lc fx.Lifecycle, sub *shrexsub.PubSub) error {
+				lc.Append(fx.Hook{
+					OnStart: sub.Start,
+					OnStop:  sub.Stop,
+				})
+				return nil
```

Comment:
I was really confused why shrexsub.Pubsub is only started for BN here and not for FN, when it should work for both.  Looked for changes it used to be the same before this PR, so I went deeper. Turned out shrexsub is started implicitly inside peer manager. But you might think that peer manager is provided to BN and FN, so whats the difference? It is started inside peer manager only if it is provided and for BN we do not provide it into peer-manager 🎉 
Really confusing and worth refactoring in separate PR.